internal/backends/qt/qt_accessible.rs

```diff
@@ -0,0 +1,699 @@
+    };
+
+    // ------------------------------------------------------------------------------
+    // Slint_accessible_item:
+    // ------------------------------------------------------------------------------
+
+    class Slint_accessible_item : public Slint_accessible, public QAccessibleValueInterface {
+    public:
+        Slint_accessible_item(void *item, QObject *obj, QAccessible::Role role, QAccessibleInterface *parent) :
+            Slint_accessible(obj, role, parent)
+        {
+            m_data = rust!(Slint_accessible_item_ctor [this: *mut c_void as "Slint_accessible_item*",
+                    item: &ItemWeak as "void*"] ->
+                    *mut SlintAccessibleItemData as "void*" {
+                        let data = SlintAccessibleItemData::new_pin_box(this, item);
+                        unsafe { Box::into_raw(Pin::into_inner_unchecked(data)) }
+            });
+        }
+
+        ~Slint_accessible_item() {
+            rust!(Slint_accessible_item_dtor [m_data: *mut SlintAccessibleItemData as "void*"] {
+                unsafe { Pin::new_unchecked(Box::from_raw(m_data)) };
+            });
+        }
+
+        void *rustItem() const override {
+            return rust!(Slint_accessible_item_rustItem [m_data: Pin<&SlintAccessibleItemData> as "void*"] -> *const ItemWeak as "void*" {
+                &m_data.item
+            });
+        }
+
+        QWidget *qwidget() const override {
+            return dynamic_cast<Slint_accessible *>(parent())->qwidget();
+        }
+
+        void *data() const {
+            return m_data;
+        }
+
+        QWindow *window() const override {
+            return parent()->window();
+        }
+
+        void delegateFocus() const override {
+            if (!has_focus) { return; }
+
+            auto index = rust!(Slint_accessible_item_delegate_focus [m_data: Pin<&SlintAccessibleItemData> as "void*"] -> i32 as "int" {
+                m_data.item.upgrade()
+                    .map(|i| { i.accessible_string_property(AccessibleStringProperty::DelegateFocus) })
+                    .and_then(|s| s.as_str().parse::<i32>().ok()).unwrap_or(-1)
+            });
+
+            if (index >= 0 && index < childCount()) {
+                static_cast<Slint_accessible_item*>(child(index))->sendFocusChangeEvent();
+            } else {
+                sendFocusChangeEvent();
+            }
+        }
+
+        // properties and state
+        QString text(QAccessible::Text t) const override {
+            return item_string_property(m_data, t);
+        }
+
+        QAccessible::State state() const override {
+            auto checked = item_string_property(m_data, CHECKED);
+
+            QAccessible::State state;
+            state.active = 1;
+            state.focusable = 1;
+            state.focused = has_focus_delegation;
+            state.checked = (checked == "true") ? 1 : 0;
+            state.checkable = checked.isEmpty() ? 0 : 1;
+            return state; /* FIXME */
```

Comment:
what's this FIXME about?

Reply:
There are about 40 flags in this struct in Qt. We will need to expose quite a lot more state to fill in all of those.